lib/updater.rb

```diff
@@ -9,19 +9,20 @@ def initialize(user_interface, database)
   attr_reader :user_interface, :database
 
   def run
+    return user_interface.display_no_contacts_message, user_interface.continue if database.database_empty?
+
     loop do
       contact_index = user_interface.choose_contact(database.all)
-      contact = database.contact_at(contact_index)
-
-      edit_contact(contact_index, contact)
+      edit_contact(contact_index)
       break unless user_interface.update_another_contact?
     end
   end
 
   private
 
-  def edit_contact(contact_index, contact)
+  def edit_contact(contact_index)
     loop do
+      contact = database.contact_at(contact_index)
```

Comment:
don't do this inside of the loop, otherwise you are pulling data from the database in every iteration, which may have performance issues.

You can save the contact outside of the loop, and then use the same copy in every iteration.